packages/runtime/container-runtime/src/summary/runningSummarizer.ts

```diff
@@ -272,94 +268,73 @@ export class RunningSummarizer extends TypedEventEmitter<ISummarizerEvents> impl
-							},
-							error,
-						);
-					}),
-				() => {},
-			);
-			refSequenceNumber++;
-		}
-		return refSequenceNumber;
+					summaryRefSeq: refSequenceNumber,
+					summaryLogger,
+				}).catch(async (error) => {
+					// If the error is 404, so maybe the fetched version no longer exists on server. We just
+					// ignore this error in that case, as that means we will have another summaryAck for the
+					// latest version with which we will refresh the state. However in case of single commit
+					// summary, we might me missing a summary ack, so in that case we are still fine as the
+					// code in `submitSummary` function in container runtime, will refresh the latest state
+					// by calling `prefetchLatestSummaryThenClose`. We will load the next summarizer from the
+					// updated state and be fine.
+					const isIgnoredError =
+						isFluidError(error) &&
+						error.errorType === DriverErrorTypes.fileNotFoundOrAccessDeniedError;
+
+					summaryLogger.sendTelemetryEvent(
+						{
+							eventName: isIgnoredError
+								? "HandleSummaryAckErrorIgnored"
+								: "HandleLastSummaryAckError",
+							referenceSequenceNumber: refSequenceNumber,
+							proposalHandle: summaryOpHandle,
+							ackHandle: summaryAckHandle,
+						},
+						error,
+					);
+				}),
+			() => {},
+		);
 	}
 
 	/**
 	 * Responsible for receiving and processing all the summaryAcks.
-	 * In case there was a summary ack processed by the running summarizer before processIncomingSummaryAcks is called,
-	 * it will wait for the summary ack that is newer than the one indicated by the lastAckRefSeq.
-	 * @param lastAckRefSeq - Identifies the minimum reference sequence number the summarizer needs to wait for.
-	 * In case of a negative number, the summarizer will wait for ANY summary ack that is greater than the deltaManager's initial sequence number,
-	 * and, in case of a positive one, it will wait for a summary ack that is greater than this current reference sequence number.
+	 * It starts processing ACKs after the one for the summary this client loaded from (initialSequenceNumber). Any
+	 * ACK before that is not interesting as it will simply be ignored.
 	 */
-	private async processIncomingSummaryAcks(lastAckRefSeq: number) {
-		let refSequenceNumber =
-			lastAckRefSeq > 0 ? lastAckRefSeq : this.runtime.deltaManager.initialSequenceNumber;
+	private async processIncomingSummaryAcks() {
+		// Start waiting for acks that are for summaries newer that the one this client loaded from.
+		let referenceSequenceNumber = this.runtime.deltaManager.initialSequenceNumber + 1;
 		while (!this.disposed) {
-			const summaryLogger = this.tryGetCorrelatedLogger(refSequenceNumber) ?? this.mc.logger;
-
-			// Initialize ack with undefined if exception happens inside of waitSummaryAck on second iteration,
-			// we record undefined, not previous handles.
-			await this.summaryCollection.waitSummaryAck(refSequenceNumber);
-
-			summaryLogger.sendTelemetryEvent({
```

Comment:
Removed this log because it doesn't add any value. Container runtime already logs in this flow and that's the one we look at and has more information.

Reply:
And there's no point of failure between this log location and container runtime log location? If there was a point of failure, this log could prove useful in debugging scenarios where the code-flow was successful till this point, but failed later on.

Reply:
No, there isn't.